src/settings.ts

```diff
@@ -245,12 +245,30 @@ export function load(): ISettings {
     };
 }
 
-export async function change(settingName: string, newValue: any, global: boolean = false): Promise<void> {
-    const configuration: vscode.WorkspaceConfiguration =
-        vscode.workspace.getConfiguration(
-            utils.PowerShellLanguageId);
+export async function getConfigurationTarget(settingName: string): Promise<vscode.ConfigurationTarget> {
+    const configuration = vscode.workspace.getConfiguration(utils.PowerShellLanguageId);
+
+    const detail = configuration.inspect(settingName);
+    let configurationTarget = null;
+    if (detail.workspaceFolderValue !== undefined) {
+        configurationTarget = vscode.ConfigurationTarget.WorkspaceFolder
+    }
+    else if (detail.workspaceValue !== undefined) {
+        configurationTarget = vscode.ConfigurationTarget.Workspace
+    }
+    else if (detail.globalValue !== undefined) {
+        configurationTarget = vscode.ConfigurationTarget.Global
+    }
+    return configurationTarget;
+}
+
+export async function change(
```

Comment:
this function is functionally still the same , only the `configurationTarget` parameter has been optimized to match exactly the called VS-Code API underneath.